document/src/main/java/com/yahoo/document/json/JsonSerializationHelper.java

```diff
@@ -116,11 +116,10 @@ public static void serializeReferenceField(JsonGenerator generator, FieldBase fi
     }
 
     public static void serializeStringField(JsonGenerator generator, FieldBase field, StringFieldValue value) {
-        // Hide empty strings
-        if (value.getString().length() == 0) {
+        // Hide fields which only contains an empty string
+        if (value.getString().length() == 0 && field != null) {
             return;
```

Comment:
I think I need some context on this one.

Reply:
When serializing an entry in an array 'field' will be null. Hence giving correct behavior for arrays.

Reply:
That context should be in the code ...